field-agent-ui/src/components/FieldAgent.js

```diff
@@ -0,0 +1,283 @@
+import { useState } from 'react';
+import data from "../data";
+import logo from "../images/field-agent-logo.png"
+
+function FieldAgents(){
+    const [agents, setAgents] = useState(data);
+    const [firstName, setFirstName] = useState('');
+    const [middleName, setMiddleName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [dob, setDob] = useState('');
+    const [heightInInches, setHeightInInches] = useState('');
+    const [editAgentId, setEditAgentId] = useState(-1);
+    const [deleteAgent, setDeleteAgent] = useState(-1);
+    const [currentView, setCurrentView] = useState('List'); // List, Add, Edit, Delete
+
+    const firstNameOnChangeHandler = (event) => {
+        console.log(event);
+        setFirstName(event.target.value);
+    }
+
+    const middleNameOnChangeHandler = (event) => {
+        console.log(event);
+        setMiddleName(event.target.value);
+    }
+
+    const lastNameOnChangeHandler = (event) => {
+        setLastName(event.target.value);
+    }
+
+    const dobNameOnChangeHandler = (event) => {
+        setDob(event.target.value);
+    }
+
+    const heightOnChangeHandler = (event) => {
+        setHeightInInches(event.target.value);
+    }
+
+    const addAgentFormSubmitHandler = (event) => {
+        event.preventDefault();
+
+        const nextId = agents.length > 0 ? Math.max(...agents.map(a => a.agentId)) + 1 : 1;
+
+        const newAgent = {
+            agentId: nextId,
+            firstName,
+            middleName,
+            lastName,
+            dob,
+            heightInInches
+        }
+
+        setAgents([...agents, newAgent]);
+
+        setFirstName('');
+        setMiddleName('');
+        setLastName('');
+        setDob('');
+        setHeightInInches('');
+
+        setCurrentView('List');
+    }
+
+    const editAgentFormSubmitHandler = (event) => {
+        event.preventDefault();
+
+        const newAgent = {
+            agentId: editAgentId,
+            firstName,
+            middleName,
+            lastName,
+            dob,
+            heightInInches
+        };
+
+        const newAgents = [...agents];
+
+        const newAgentIndex = newAgents.findIndex(agent => agent.agentId === editAgentId);
+
+        newAgents[newAgentIndex] = newAgent;
+
+        setAgents(newAgents);
+
+        setFirstName('');
+        setMiddleName('');
+        setLastName('');
+        setDob('');
+        setHeightInInches('');
+
```

Comment:
Since this code is used more than once, I would set it in a separate resetForm() function and pop that in these handlers.